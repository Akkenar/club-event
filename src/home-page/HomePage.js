import React, { useEffect } from 'react';
import InformationSection from '../information-section/InformationSection';
import { setPageTitle } from '../page.lib';
import getKey from '../intl/getKey';
import { injectIntl } from 'react-intl';

const HomePage = props => {
  useEffect(() => {
    setPageTitle(getKey('home.page.title', props));
  });

  return (
    <div>
      <InformationSection />
    </div>
  );
};

export default injectIntl(HomePage);
