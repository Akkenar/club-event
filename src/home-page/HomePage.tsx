import React from 'react';
import { Fragment, useContext, useEffect } from 'react';
import { Header } from 'semantic-ui-react';
import imgTop from '../assets/Arnaud-Conne-2017_5.jpg';
import getKey from '../core/intl/getKey';
import LanguageContext from '../core/intl/LanguageContext';
import LazyImage from '../core/lazy-image/LazyImage';
import Logo from '../core/logo/Logo';
import { goToTop, setPageTitle } from '../core/page.lib';
import InformationSection from './information-section/InformationSection';
import ScrollableAnchor from '../core/scrollable-anchor/ScrollableAnchor';
import MapIframe from './MapIframe';
import ScheduleSection from './schedule-section/ScheduleSection';
import SponsorsSection from './sponsors-section/SponsorsSection';
import RegisterSection from './register-section/RegisterSection';
import PageMenu from './PageMenu';

const HomePage = () => {
  const { messages } = useContext(LanguageContext);

  useEffect(() => {
    setPageTitle(getKey('home.page.title', messages));
    goToTop();
  });

  return (
    <Fragment>
      <PageMenu />
      <Header as="h1" data-testid="main-title">
        <ScrollableAnchor text={getKey('home.page.title', messages)} />
      </Header>
      <Logo big={true} width={320} height={232} />
      <InformationSection />
      <MapIframe />
      <ScheduleSection />
      <SponsorsSection />
      <RegisterSection />
      <LazyImage
        width={320}
        height={212}
        src={imgTop}
        alt={'Photo: Arnaud Conne'}
      />
    </Fragment>
  );
};

export default HomePage;
