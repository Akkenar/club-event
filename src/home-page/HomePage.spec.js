import React from 'react';
import { render } from 'react-testing-library';
import HomePage from './HomePage';

describe('HomePage', () => {
  it('should match the snapshot', () => {
    const wrapper = render(<HomePage />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should set the page title', () => {
    // Will be set by side effect through a hook
    render(<HomePage />);
    expect(document.title).toEqual('home.page.title');
  });
});
