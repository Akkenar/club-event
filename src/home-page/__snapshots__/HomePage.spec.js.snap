// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match the snapshot 1`] = `
<body>
  <div>
    <h1
      class="ui header"
    >
      home.page.title
    </h1>
    <img
      alt="logo.alt"
      class="Logo  Logo--big"
      src="test-file-stub"
      title="logo.alt"
    />
    <div />
    <iframe
      allowfullscreen=""
      frameborder="0"
      height="600"
      src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2735.705437069773!2d6.377488315962078!3d46.71154597913546!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x478db3ee662c90db%3A0x172582dbb0e2b277!2sAuberge+Communale+de+Vallorbe!5e0!3m2!1sen!2sch!4v1541839403749"
      style="border: 0px; margin: 1em 0px;"
      title="Vallorbe"
      width="100%"
    />
    <h2
      class="ui header"
    >
      home.page.register
    </h2>
    <div
      class="ui warning message"
    >
      home.page.startofregistration
    </div>
    <svg
      aria-label="Photo: Arnaud Conne"
      class="LazyImage"
      viewBox="0 0 1024 1024"
      width="320"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d=""
      />
    </svg>
  </div>
</body>
`;
